src/frontend/src/handler/create_table.rs

```diff
@@ -1845,23 +1858,51 @@ pub async fn generate_stream_graph_for_replace_table(
     _session: &Arc<SessionImpl>,
     table_name: ObjectName,
     original_catalog: &Arc<TableCatalog>,
-    format_encode: Option<FormatEncodeOptions>,
     handler_args: HandlerArgs,
+    statement: Statement,
     col_id_gen: ColumnIdGenerator,
-    column_defs: Vec<ColumnDef>,
-    wildcard_idx: Option<usize>,
-    constraints: Vec<TableConstraint>,
-    source_watermarks: Vec<SourceWatermark>,
-    append_only: bool,
-    on_conflict: Option<OnConflict>,
-    with_version_column: Option<String>,
-    cdc_table_info: Option<CdcTableInfo>,
-    include_column_options: IncludeOption,
-    engine: Engine,
     sql_column_strategy: SqlColumnStrategy,
 ) -> Result<(StreamFragmentGraph, Table, Option<PbSource>, TableJobType)> {
     use risingwave_pb::catalog::table::OptionalAssociatedSourceId;
 
+    let Statement::CreateTable {
+        columns,
+        constraints,
+        source_watermarks,
+        append_only,
+        on_conflict,
+        with_version_column,
+        wildcard_idx,
+        cdc_table_info,
+        format_encode,
+        include_column_options,
+        engine,
+        with_options,
+        ..
+    } = statement
+    else {
+        panic!("unexpected statement type: {:?}", statement);
+    };
+
+    let format_encode = format_encode
+        .clone()
+        .map(|format_encode| format_encode.into_v2_with_warning());
+
+    let engine = match engine {
+        risingwave_sqlparser::ast::Engine::Hummock => Engine::Hummock,
+        risingwave_sqlparser::ast::Engine::Iceberg => Engine::Iceberg,
+    };
+
+    let is_drop_connector = {
+        original_catalog.associated_source_id().is_some()
+            && format_encode.is_none()
```

Comment:
I think `original...is_some() && format_encode.is_none()` should be sufficient? Shall we turn the rest into assertions?